File: 8-kyu/clojure/remove-even-list-items.clj
;; Take an array and remove every second element out of that array. Always keep the first element and start removing with the next element.

;; Example: arr = ['Keep', 'Remove', 'Keep', 'Remove', 'Keep', ...];

;; None of the arrays will be empty, so you don't have to worry about that!

(ns kata)

(defn remove-every-other [arr]
  (keep-indexed #(fn [index item] (when (= (mod index 2) 0) item)) arr))

File: 8-kyu/js/remove-even-list-items.js
/* Take an array and remove every second element out of that array. Always keep the first element and start removing with the next element.

Example: arr = ['Keep', 'Remove', 'Keep', 'Remove', 'Keep', ...];

None of the arrays will be empty, so you don't have to worry about that! */

function removeEveryOther(arr) {
  const newArr = [];
  for (let i = 0; i < arr.length; i += 2) {
    newArr.push(arr[i]);
  }

  return newArr;
}
